google/cloud/storage/benchmarks/storage_shard_throughput_benchmark.cc

```diff
@@ -0,0 +1,481 @@
+    std::cout << r.bytes << ',' << r.elapsed.count() << std::endl;
+    auto const MiB = r.bytes / gcs_bm::kMiB;
+    auto const MiBs =
+        MiB * (1.0 * decltype(r.elapsed)::period::den) / r.elapsed.count();
+    MiBs_sum += MiBs;
+  }
+
+  auto const MiBs_avg = MiBs_sum / options->iteration_count;
+  std::cout << "# Average Bandwidth (MiB/s): " << MiBs_avg << "\n";
+
+  DeleteAllObjects(client, bucket_name, *options, object_names);
+
+  std::cout << "# Deleting " << bucket_name << "\n";
+  auto status = client.DeleteBucket(bucket_name);
+  if (!status.ok()) {
+    std::cerr << "# Error deleting bucket, status=" << status << "\n";
+    return 1;
+  }
+
+  return 0;
+}
+
+namespace {
+
+void CreateGroup(gcs::Client client, std::string const& bucket_name,
+                 Options const& options, std::vector<std::string> group) {
+  google::cloud::internal::DefaultPRNG generator =
+      google::cloud::internal::MakeDefaultPRNG();
+
+  std::string const random_data =
+      gcs_bm::MakeRandomData(generator, options.chunk_size);
+  for (auto const& object_name : group) {
+    auto stream = client.WriteObject(bucket_name, object_name, gcs::Fields(""));
+    for (std::int64_t count = 0; count != options.chunk_count; ++count) {
+      stream.write(random_data.data(), random_data.size());
+    }
+    stream.Close();
+    if (!stream.metadata()) {
+      std::cerr << "Error writing: " << object_name << "\n";
+    }
+  }
+}
+
+std::vector<std::string> CreateAllObjects(
+    gcs::Client client, google::cloud::internal::DefaultPRNG& gen,
+    std::string const& bucket_name, Options const& options) {
+  using std::chrono::duration_cast;
+  using std::chrono::milliseconds;
+
+  std::size_t const max_group_size =
+      std::max(options.object_count / options.thread_count, 1);
+  std::cout << "# Creating test objects [" << max_group_size << "]\n";
+
+  // Generate the list of object names.
+  std::vector<std::string> object_names;
+  object_names.reserve(options.object_count);
+  for (long c = 0; c != options.object_count; ++c) {
+    object_names.emplace_back(gcs_bm::MakeRandomObjectName(gen));
+  }
+
+  // Split the objects in more or less equally sized groups, launch a thread
+  // to create the objects in each group.
+  auto start = std::chrono::steady_clock::now();
+  std::vector<std::future<void>> tasks;
+  std::vector<std::string> group;
+  for (auto const& o : object_names) {
+    group.push_back(o);
+    if (group.size() >= max_group_size) {
+      tasks.emplace_back(std::async(std::launch::async, &CreateGroup, client,
+                                    bucket_name, options, std::move(group)));
+      group = {};  // after a move, must assign to guarantee it is valid.
+    }
+  }
+  if (!group.empty()) {
+    tasks.emplace_back(std::async(std::launch::async, &CreateGroup, client,
+                                  bucket_name, options, std::move(group)));
+    group = {};  // after a move, must assign to guarantee it is valid.
```

Comment:
This one seems unnecessary, and therefore a little misleading.

Reply:
Removed.